lib/fluent/plugin/out_bigquery.rb

```diff
@@ -382,115 +311,177 @@ def format(tag, time, record)
         buf = String.new
         row = @fields.format(@add_time_field.call(record, time))
         unless row.empty?
-          row = {"json" => row}
-          row['insert_id'] = @get_insert_id.call(record) if @get_insert_id
-          buf << row.to_msgpack
+          buf << MultiJson.dump(row) + "\n"
         end
         buf
       end
 
-      def _write(chunk, table_format, template_suffix_format)
-        rows = []
-        chunk.msgpack_each do |row_object|
-          # TODO: row size limit
-          rows << row_object.deep_symbolize_keys
+      def write(chunk)
+        table_id_format = @tables_mutex.synchronize do
+          t = @tables_queue.shift
+          @tables_queue.push t
+          t
         end
+        template_suffix_format = @template_suffix
+        _write(chunk, table_id_format, template_suffix_format)
+      end
 
-        now = Time.at(Fluent::Engine.now)
-        group = rows.group_by do |row|
-          [
-            generate_table_id(table_format, now, row, chunk),
-            template_suffix_format ? generate_table_id(template_suffix_format, now, row, chunk) : nil,
-          ]
-        end
-        group.each do |(table_id, template_suffix), group_rows|
-          insert(table_id, group_rows, template_suffix)
-        end
+      # for delayed commit
+      def try_write(chunk)
```

Comment:
Right - it's not good as an example, because it has a special role as testbed of `#write`/`#try_write` implementations.